### PR TITLE
feat: introduces a pre-commit checker for secrets and potential private keys

### PR DESCRIPTION
**Pull request by 0xDEnYO**
# Which Jira task belongs to this PR?

# Why did I implement it this way?

# Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] I have updated any required documentation
- [ ] If this requires a contract version change, I have updated the version number in the source file

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

**Review by ezynda3 — APPROVED** [review]

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,4 +1,80 @@
...
+# Load secrets from .env file
+if [ -f ".env" ]; then
+  ENV_SECRETS=$(grep -v '^#' .env | grep -v '^\s*$' | cut -d '=' -f 2- | sed 's/ *#.*//')
+else
+  echo ".env file not found"
+  ENV_SECRETS=""
+fi
+
+printAdvise() {
+  ADVISE="If your commit did contain a secret, undo the commit with 'git reset soft HEAD~1', remove the private key and commit again"
```
Comment (maxklenk) [inline .husky/pre-commit:37]: Please adjust command to `git reset --soft HEAD~1`

Comment (maxklenk) [reply .husky/pre-commit:37]: Do show another message if the commit was aborted, as in this case no commit was added, so a reset is not needed.

**Review by maxklenk — CHANGES_REQUESTED** [review]
Thanks for adding. The detection works fine.

File: .husky/pre-commit (hunk 1 above)
Comment (0xDEnYO) [reply .husky/pre-commit]: both done

**Review by maxklenk — APPROVED** [review]

**Pull request merged**